### PR TITLE
-luadev flag and lua log filter

### PR DESCRIPTION
**Pull request by EatThePath**
Adds new default 'lua' filter category for all lua errors and output and -luadev flag that currently only makes lua errors non-fatal.

### CODE REVIEW

**Comment by EatThePath** [comment]
Implements two points of #4665

File: code/cmdline/cmdline.cpp [hunk 1]
```diff
@@ -512,6 +512,7 @@ cmdline_parm graphics_debug_output_arg("-gr_debug", nullptr, AT_NONE); // Cmdlin
 cmdline_parm log_to_stdout_arg("-stdout_log", nullptr, AT_NONE); // Cmdline_log_to_stdout
 cmdline_parm slow_frames_ok_arg("-slow_frames_ok", nullptr, AT_NONE);	// Cmdline_slow_frames_ok
 cmdline_parm fixed_seed_rand("-seed", nullptr, AT_INT);	// Cmdline_rng_seed,Cmdline_reuse_rng_seed;
+cmdline_parm luadev_arg("-luadev", "nullptr", AT_NONE);	// Cmdline_slow_frames_ok
```
Comment (BMagnu) [inline code/cmdline/cmdline.cpp:515]: That comment seems slightly outdated...

**Review by BMagnu — CHANGES_REQUESTED** [review]
Mostly good.
Also, do I correctly understand this, that it will keep printing lua to the debug log without any special logging config?

File: code/cmdline/cmdline.cpp (hunk 1 above)
Comment (EatThePath) [reply code/cmdline/cmdline.cpp:515]: Whoops, I'll update that...

**Comment by EatThePath** [comment]
> Mostly good. Also, do I correctly understand this, that it will keep printing lua to the debug log without any special logging config?

Yeah, I tested to make sure that it defaults to lua on if not specified.

**Review by BMagnu — APPROVED** [review]

**Pull request merged**